### PR TITLE
infra: drop support for Python 3.8 and add Python 3.11 builds

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10"]
```
Comment (speller26) [inline .github/workflows/python-package.yml:22]: Why not add 3.11, which we support in our other repos?

File: .github/workflows/conda-build.yml [hunk 2]
```diff
@@ -24,10 +24,10 @@ jobs:
         sudo rm -rf /usr/share/dotnet
         sudo rm -rf /opt/ghc
     - uses: actions/checkout@v2
-    - name: Setup python 3.8
+    - name: Setup python 3.10
       uses: actions/setup-python@v2
       with:
-        python-version: 3.8
+        python-version: 3.10.13
```
Comment (speller26) [inline .github/workflows/conda-build.yml:30]: Why no just go ahead with 3.11?

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (math411) [reply .github/workflows/python-package.yml:22]: Good call

File: .github/workflows/conda-build.yml (hunk 2 above)
Comment (math411) [reply .github/workflows/conda-build.yml:30]: Adding that

File: examples/hybrid_quantum_algorithms/QAOA/utils_classical.py [hunk 3]
```diff
@@ -6,8 +6,6 @@
 import networkx as nx
 from pyqubo import Spin, solve_ising
 
-warnings.filterwarnings("ignore", category=matplotlib.cbook.mplDeprecation)
```
Comment (krneta) [inline examples/hybrid_quantum_algorithms/QAOA/utils_classical.py:9]: Why did you remove this?

**Review by krneta — APPROVED** [review]

File: examples/hybrid_quantum_algorithms/QAOA/utils_classical.py (hunk 3 above)
Comment (math411) [reply examples/hybrid_quantum_algorithms/QAOA/utils_classical.py:9]: This is no longer needed: https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.7.0.html#deprecation-aliases-in-cbook
